dpe/src/commands/mod.rs

```diff
@@ -67,26 +68,56 @@ impl Command {
     ///
     /// * `bytes` - serialized command
     pub fn deserialize(bytes: &[u8]) -> Result<Command, DpeErrorCode> {
-        let cmd_header = CommandHdr::try_from(bytes)?;
+        let header = CommandHdr::read_from_prefix(bytes).ok_or(DpeErrorCode::InvalidArgument)?;
+        if header.magic != CommandHdr::DPE_COMMAND_MAGIC {
+            return Err(DpeErrorCode::InvalidCommand);
+        }
+        // The client doesn't know what profile is implemented when calling the `GetProfile`
+        // command. But, all other commands should be directed towards the correct profile.
+        if header.cmd_id != Command::GET_PROFILE && header.profile != DPE_PROFILE as u32 {
+            return Err(DpeErrorCode::InvalidCommand);
+        }
```

Comment:
Do we believe we need to parse a `CommandHdr` somewhere else than in this `Command::deserialize()` function? If yes and if those magic and profile checks are always expected to occur, then we should probably keep this as a separate function like it was the case before.

Reply:
I can't think of another place, but it also fits a lot better in the original location. I put the whole thing back for `CommandHdr` including taking out the derive. I didn't want a similar scenario where someone didn't realize there was special logic in `CommandHdr::try_from` that wouldn't be in `CommandHdr::read_from_prefix` and misuse it.

Reply:
I'm not convinced about the "taking out the derive" part. Why not have a `TryFrom` implementation that uses the derive?

```rust
impl TryFrom<&[u8]> for CommandHdr {
    type Error = DpeErrorCode;

    fn try_from(raw: &[u8]) -> Result<Self, Self::Error> {
        let header = CommandHdr::read_from_prefix(raw).map_err(|_| DpeErrorCode::InvalidCommand)?;
        if header.magic != Self::DPE_COMMAND_MAGIC {
            return Err(DpeErrorCode::InvalidCommand);
        }
        // The client doesn't know what profile is implemented when calling the `GetProfile`
        // command. But, all other commands should be directed towards the correct profile.
        if header.cmd_id != Command::GET_PROFILE && header.profile != DPE_PROFILE as u32 {
            return Err(DpeErrorCode::InvalidCommand);
        }
        Ok(header)
    }
}
```

Reply:
My concern is that someone later on could use `read_from_prefix` elsewhere and not realize they were missing key checks that exist in `try_from`. Ideally, if we could make the `derive` private that would solve the concern.

I don't think its too major of a concern though, I put in your suggestion.

Reply:
I think (1) is good. It's pretty low risk so its probably not worth making too complicated of a solution. We can always change it later if we need to.